lib/problems.rb

```diff
@@ -1,9 +1,36 @@
 require_relative './stack.rb'
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n), will always have to traverse through the string
+# Space Complexity: O(n), because in the worst case all char in string
+# are opening brackets, best case would be O(1) where we push and pop every item
 def balanced(string)
```

Comment:
👍 , However you can dry up the case statement by using a hash instead.  The has would have the open braces as the keys and the closing braces as the values.

```ruby
return false if brace_hash[current_char] != char
```